compiler/testData/diagnostics/tests/incompleteCode/inExpr.fir.kt

```diff
@@ -1,5 +0,0 @@
-package l
-
-fun test(a: Int) {
-    if (a <!UNRESOLVED_REFERENCE!>in<!><!SYNTAX!><!> ) {} //a is not unresolved
```

Comment:
This might be aligned with my other comment.

So, the reason `in` is unresolved is due to the lack of arguments (or their types). It makes sense (to me) if we decide to _not_ report `UNRESOLVED_REFERENCE` since its root cause is something else (`SYNTAX`) inside (i.e., children level). But, currently, IIUC, this is discarded while filtering `source`, which I'm not sure is the right way or matches the way FE 1.0 reports the kinda root cause.

Reply:
I think we should definitely still try to do more filtering and only report the root cause whenever possible like you suggested in a previous comment.

On the other hand, I don't see anything wrong with leveraging this side effect of missing reporting target to achieve this goal for these particular situations.